index.js

```diff
@@ -342,31 +342,133 @@
         /**
          * PropTypes
          */
-        
-        function observableTypeChecker (type) {
-            return function(props, propName, componentName) {
-                if (!mobx['isObservable' + type](props[propName])) {
+
+        // Copied from React.PropTypes
+        function createChainableTypeChecker(validate) {
+            function checkType(isRequired, props, propName, componentName, location, propFullName) {
+                componentName = componentName || ANONYMOUS;
+                propFullName = propFullName || propName;
+                if (props[propName] == null) {
+                    if (isRequired) {
+                        var actual = props[propName] === null ? 'null' : 'undefined';
+                        return new Error(
+                          'The ' + location + ' `' + propFullName +'` is marked as required ' +
+                          'in `' + componentName + '`, but its value is `' + actual + '`.'
+                        );
+                    }
+                    return null;
+                } else {
+                    return validate(props, propName, componentName, location, propFullName);
+                }
+            }
+            var chainedCheckType = checkType.bind(null, false);
+            chainedCheckType.isRequired = checkType.bind(null, true);
+            return chainedCheckType;
+        }
+
+        // Copied from React.PropTypes
+        function isSymbol(propType, propValue) {
+            // Native Symbol.
+            if (propType === 'symbol') {
+                return true;
+            }
+
+            // 19.4.3.5 Symbol.prototype[@@toStringTag] === 'Symbol'
+            if (propValue['@@toStringTag'] === 'Symbol') {
+                return true;
+            }
+
+            // Fallback for non-spec compliant Symbols which are polyfilled.
+            if (typeof Symbol === 'function' && propValue instanceof Symbol) {
+                return true;
+            }
+
+            return false;
+        }
+
+        // Copied from React.PropTypes
```

Comment:
Are you ok about building it with browserify? That would enable us to use https://github.com/mridgway/hoist-non-react-statics for #92